scripts/pylib/twister/twisterlib/twister_main.py

```diff
@@ -160,7 +160,11 @@ def main(options):
         report.json_report(options.save_tests)
         return 0
 
-    if options.device_testing and not options.build_only:
+    if options.report_summary:
+        report.synopsis()
+        return 0
+
+    if options.device_testing and (not options.build_only or not options.report_summary):
```

Comment:
the option is not effective here assuming the script exits few lines before

```suggestion
    if options.device_testing and (not options.build_only):
```